### PR TITLE
[ISSUE #5292] [RIP-48] Support reset offset in server-side to improve the success rate

### PR DESCRIPTION
**Pull request by lizhimins**

### CODE REVIEW

**Comment by lizhimins** [comment]
Please help review this pr

File: common/src/main/java/org/apache/rocketmq/common/protocol/header/ResetOffsetRequestHeader.java [hunk 1]
```diff
@@ -22,12 +22,21 @@
...
     @CFNotNull
     private String topic;
+
     @CFNotNull
     private String group;
+
+    private int queueId = -1;
+
+    @CFNotNull
+    private Long offset;
```
Comment (xdkxlk) [inline common/src/main/java/org/apache/rocketmq/common/protocol/header/ResetOffsetRequestHeader.java:35]: The CFNotNull of offset is not necessary, and will produce WARN for old client

Comment (lizhimins) [reply common/src/main/java/org/apache/rocketmq/common/protocol/header/ResetOffsetRequestHeader.java:34]: ok, I have fixed it

**Comment by lizhanhui** [comment]
Rebase to fix file conflicts

**Review by lizhanhui — APPROVED** [review]

**Pull request merged**